code/controllers/subsystem/mapping.dm

```diff
@@ -375,6 +375,17 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 	else
 		for(var/M in global.config.maplist)
 			mapvotes[M] = 1
+			
+	var/list/previous_maps = list()
+	if(SSdbcore.Connect())
+		var/datum/DBQuery/query_previous_maps = SSdbcore.NewQuery({"
+			SELECT map_name FROM [format_table_name("round")] WHERE id BETWEEN lower = :lower AND upper = :upper
+		"}, list("lower" = GLOB.round_id - 9, "upper" = GLOB.round_id))
+		if(!query_previous_maps.Execute())
+			qdel(query_previous_maps)
+		while(query_previous_maps.NextRow())
+			previous_maps[query_previous_maps.item[1]] += 1
+		qdel(query_previous_maps)
```

Comment:
Double qdel if it failed

Reply:
Does qdeling an object that is already queued to be deleted have consequences?

Reply:
Generally throws a runtime from what ive seen